gitindex/index.go

```diff
@@ -505,6 +506,17 @@ func indexGitRepo(opts Options, config gitIndexConfig) error {
 		if err != nil {
 			return err
 		}
+
+		// Compute the mean rank for this repository. Note: we overwrite the rank
```

Comment:
This got a little messy -- the ranking service returns the global mean rank (across all repos), but I found it was important to use the mean rank for each repo. I filed https://github.com/sourcegraph/sourcegraph/issues/49060 to track all our clean-ups so I don't forget!